src/main/java/seedu/address/model/transaction/TxnStatus.java

```diff
@@ -0,0 +1,67 @@
+package seedu.address.model.transaction;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.AppUtil.checkArgument;
+
+import java.time.LocalDateTime;
+/**
+ * Status of Transaction
+ */
+public class TxnStatus {
+
+    public static final String MESSAGE_CONSTRAINTS =
+            "Transaction status should only be open or closed, and it should not be blank";
+
+    /* The input should be "open" or "closed" */
+    public static final String VALIDATION_REGEX = ".*\\bopen\\b|.*\\bclosed\\b";
```

Comment:
Along with the `isClosed()` method on line 36, it might be clearer if we define static final variables for 'Open' and 'Closed' (such as `CLOSED_TXN` and `OPEN_TXN`), and reference them within the class.

Might also want to consider changing the validation regex to private, since it validation of a TxnStatus is done via the public `isValidTxnStatus()` method.

Reply:
"changing the validation regex to private" -- I think VALIDATION_REGEX is declared public in all class files, so can keep it for consistency unless we change all of them.

Noted on the static final variables.